canvas/src/layer/layer_2d.rs

```diff
@@ -3,32 +3,161 @@ use crate::{
 };
+
 use std::{
   cmp::PartialEq,
   ops::{Deref, DerefMut},
 };
 
+pub type Vector = euclid::default::Vector2D<f32>;
+pub type Angle = euclid::Angle<f32>;
+
 /// The 2d layer is a two-dimensional grid. The coordinate (0, 0) is at the
 /// upper-left corner of the canvas. Along the X-axis, values increase towards
 /// the right edge of the canvas. Along the Y-axis, values increase towards the
 /// bottom edge of the canvas.
-#[derive(Default, Debug, Clone)]
+// #[derive(Default, Debug, Clone)]
 pub struct Rendering2DLayer<'a> {
   state_stack: Vec<State>,
   pub(crate) commands: Vec<Command<'a>>,
+  path: Option<Builder>,
 }
 
 impl<'a> Rendering2DLayer<'a> {
   pub fn new() -> Self {
     Self {
       state_stack: vec![State::new()],
       commands: vec![],
+      path: None,
     }
   }
 
+  #[inline]
+  pub fn get_path(&mut self) -> Option<Path> { self.path.take().map(|b| b.build()) }
+
+  /// Starts a new path by emptying the list of sub-paths.
+  /// Call this method when you want to create a new path.
+  #[inline]
+  pub fn begin_path(&mut self, x: f32, y: f32) -> &mut Self {
+    self.path = Some(Builder::new());
+    self.path.as_mut().map(|b| b.begin(point(x, y)));
+    self
+  }
+
+  /// Causes the point of the pen to move back to the start of the current
+  /// sub-path. It tries to draw a straight line from the current point to the
+  /// start. If the shape has already been closed or has only one point, this
+  #[inline]
+  pub fn close_path(&mut self) -> &mut Self {
+    self.path.as_mut().map(|b| b.close());
+    self
+  }
+
+  /// Connects the last point in the current sub-path to the specified (x, y)
+  /// coordinates with a straight line.
+  #[inline]
+  pub fn line_to(&mut self, x: f32, y: f32) -> &mut Self {
+    self.path.as_mut().map(|b| b.line_to(point(x, y)));
+    self
+  }
+
+  /// Adds a cubic Bezier curve to the current path.
+  #[inline]
+  pub fn bezier_curve_to(
+    &mut self,
+    cp1x: f32,
+    cp1y: f32,
+    cp2x: f32,
+    cp2y: f32,
+    x: f32,
+    y: f32,
+  ) -> &mut Self {
+    self
+      .path
+      .as_mut()
+      .map(|b| b.cubic_bezier_to(point(cp1x, cp1y), point(cp2x, cp2y), point(x, y)));
+    self
+  }
+
+  /// Adds a quadratic Bézier curve to the current path.
+  #[inline]
+  pub fn quadratic_curve_to(&mut self, cpx: f32, cpy: f32, x: f32, y: f32) -> &mut Self {
+    self
+      .path
+      .as_mut()
+      .map(|b| b.quadratic_bezier_to(point(cpx, cpy), point(x, y)));
+    self
+  }
+
+  /// Adds a circular arc to the current path.
+  pub fn arc(
+    &mut self,
+    x: f32,
+    y: f32,
+    radius: f32,
+    start_angle: Angle,
+    end_angle: Angle,
+  ) -> &mut Self {
+    let sweep_angle = end_angle - start_angle;
+    let arc = Arc {
+      start_angle,
+      sweep_angle,
+      radii: Vector::new(radius, radius),
+      center: point(x, y),
+      x_rotation: Angle::zero(),
+    };
+    let arc_start = arc.from();
+    self.begin_path(arc_start.x, arc_start.y);
```

Comment:
When add an arc to path, should we really need call begin & close, or just add it, and wait user to close it, stroke it or fill it ?